lambeq/experimental/discocirc/reader.py

```diff
@@ -400,10 +402,11 @@ def text2circuit(self,
 
     def _calculate_reindices(self,
                              orig_toks,
-                             parsed_toks):
+                             parsed_toks,
+                             parsed_toks_indxs):
         reindexer = {}
         j = 0
-        for i, otok in enumerate(parsed_toks):
```

Comment:
@neiljdo in what situation are an enumeration and `parsed_tok_indxs` going to disagree? I thought in the tree they're guaranteed to be `[0 ... n-1]`?

Reply:
@nikhilkhatri it happens when you pass an already tokenised sentence but then bobcat removes one of the tokens, usually a punctuation, when creating the diagram that will then be used by the tree.

Also, it happens when the tree is modified after, e.g. when adding the `''` word for missing nouns. The `''` was never in the original tokens.